src/main/java/mate/academy/service/impl/MovieServiceImpl.java

```diff
@@ -14,16 +14,23 @@ public class MovieServiceImpl implements MovieService {
 
     @Override
     public Movie add(Movie movie) {
+        boolean isMovieExist = movieDao.getAll()
+                .stream()
+                .anyMatch(m -> m.getTitle().equals(movie.getTitle()));
+        if (isMovieExist) {
+            throw new RuntimeException("Movie already exists");
+        }
```

Comment:
This logic to check if a movie exists by retrieving all movies and then streaming over them is inefficient. It can lead to performance issues with a large number of movies. Consider implementing a method in the `MovieDao` that can check for the existence of a movie more efficiently, perhaps by using a query that counts the number of movies with the given title.